enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/protocol/NettyPipelineBuilder.java

```diff
@@ -145,6 +146,16 @@ public BUILDER addGate( Predicate<Object> gatePredicate )
         return self;
     }
 
+    public BUILDER onClose( Runnable closeHandler )
+    {
+        if ( this.closeHandler != null )
```

Comment:
Might be worth considering a step builder. Probably not during code freeze though.